fe1-web/src/features/witness/components/WitnessNotification.tsx

```diff
@@ -0,0 +1,120 @@
+import PropTypes from 'prop-types';
+import React, { FunctionComponent, useEffect, useMemo } from 'react';
+import { Text, View } from 'react-native';
+import { Button } from 'react-native-elements';
+import { useSelector } from 'react-redux';
+
+import { makeMessageSelector } from 'core/network/ingestion';
+import { dispatch } from 'core/redux';
+
+import { WitnessHooks } from '../hooks';
+import { WitnessFeature } from '../interface';
+import { requestWitnessMessage } from '../network/WitnessMessageApi';
+import { removeMessageToWitness } from '../reducer';
+
+const WitnessNotification = ({ notification, navigateToNotificationScreen }: IPropTypes) => {
+  const messageSelector = useMemo(
+    () => makeMessageSelector(notification.messageId),
+    [notification.messageId],
+  );
+  const message = useSelector(messageSelector);
+
+  const discardNotifications = WitnessHooks.useDiscardNotifications();
+  const markNotificationAsRead = WitnessHooks.useMarkNotificationAsRead();
+  const isEnabled = WitnessHooks.useIsEnabled();
+  const laoId = WitnessHooks.useCurrentLaoId();
+
+  // if the notification state somehow gets out of sync, remove the corresponding notification
+  useEffect(() => {
+    if (!message) {
+      dispatch(
+        discardNotifications({ laoId: laoId.valueOf(), notificationIds: [notification.id] }),
+      );
+      navigateToNotificationScreen();
+      console.warn(
+        `There was a notification with id ${notification.id} in the redux store referencing a message id (${notification.messageId}) that is not (no longer?) stored`,
+      );
+    }
+  }, [
+    laoId,
+    navigateToNotificationScreen,
+    discardNotifications,
+    notification.id,
+    notification.messageId,
+    message,
+  ]);
+
+  const onWitness = () => {
+    if (message) {
+      dispatch(
+        discardNotifications({ laoId: laoId.valueOf(), notificationIds: [notification.id] }),
+      );
+      if (isEnabled) {
+        requestWitnessMessage(message.channel, message.message_id);
+      }
+      dispatch(removeMessageToWitness(message.message_id.valueOf()));
+      navigateToNotificationScreen();
+    }
+  };
+
+  const onDecline = () => {
+    if (message) {
+      dispatch(markNotificationAsRead({ laoId: laoId.valueOf(), notificationId: notification.id }));
+      navigateToNotificationScreen();
+    }
+  };
+
+  return (
+    <View>
+      <Text>{JSON.stringify(notification)}</Text>
+      <Text>{JSON.stringify(message)}</Text>
+      <Button title="Witness Message" onPress={onWitness} />
+      <Button title="Decline Message" onPress={onDecline} />
+    </View>
```

Comment:
Is it temporary? I'm not sure that a user which sees a JSON would understand it directly.

Reply:
Yes this is temporary. (One of the reason for say `ugly` version) 😅
I am not sure yet what information should be displayed (since it also depends on the message) and what the best way of doing so is :) I thought I'd be better to improve this in a separate PR?